src/banana/toy.py

```diff
@@ -61,7 +61,8 @@ def toyLHPDFs(pid, x):  # pylint: disable=too-many-locals
     xpdf[-1] = xdbar
     xpdf[-2] = xubar
     xpdf[-3] = xsbar
-
+    if pid not in xpdf:
+        return 0.0
```

Comment:
Since we are actually never using this function, you can actually move all the setup in the constructor of the class, and the extra condition (`pid not in xpdf`) in one of the two callables (e.g. `xfxQ2`) and make the other one an alias to the chosen one, e.g.:
```py
def xfxQ(self, pid, x, _Q):
    return self.xfxQ2(pid, x)
```